core/src/main/java/org/elasticsearch/search/profile/SearchProfileShardResults.java

```diff
@@ -98,15 +99,14 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             entry.getValue().getAggregationProfileResults().toXContent(builder, params);
             builder.endObject();
         }
-
         builder.endArray().endObject();
         return builder;
     }
 
     public static SearchProfileShardResults fromXContent(XContentParser parser) throws IOException {
         XContentParser.Token token = parser.currentToken();
         ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
-        Map<String, ProfileShardResult> searchProfileResults = new HashMap<>();
+        Map<String, ProfileShardResult> searchProfileResults = new LinkedHashMap<>();
```

Comment:
This way we preseve the order of the shard keys we encountered during xContent parsing. Without preserving this order, I encountered different xContent rendering outputs for the "shards" section (which is an array) in the roundtrip tests, so the order matters (at least the way `assertToXContentEquivalent` works so far).